casepro/pods/views.py

```diff
@@ -34,4 +38,15 @@ def perform_pod_action(request, index):
     except ValueError as e:
         return JsonResponse({'reason': 'JSON decode error', 'details': e.message}, status=400)
 
-    return JsonResponse(pod.perform_action(data))
+    case_id = data.get('case_id')
+    if case_id is None:
+        return JsonResponse(
+            {'reason': 'Request object needs to have a "case_id" field'}, status=400)
+
+    action_data = data.get('action', {})
+    success, payload = pod.perform_action(action_data.get('type'), action_data.get('payload', {}))
+    if success is True:
+        case = Case.objects.get(id=case_id)
```

Comment:
Actually come to think of it, it should actually be checking `Case.access_level(user)` to make sure this user has correct access rights to this case.

Obviously this applies to anywhere requests can access or modify cases.